tests/unit/libexpr/url-name.cc

```diff
@@ -1,4 +1,4 @@
-#include "url-name.hh"
+#include "flake/url-name.hh"
```

Comment:
Should the test also be placed inside a flake directory to match the path of the implementation?

Reply:
We're currently not consistent about it. We do this for `value/`, but not for `flake/` yet. I think we should use the matching directory name, yes.

Reply:
Good point. I moved it and added the new `flake/` directory to to the makefile. I can also move all the other flake tests if you want, though it might be out of scope for this PR.

Reply:
Ok there was only one more flake-related test in libexpr, I moved that as well.